src/main/java/com/leveluptasks/entity/User.java

```diff
@@ -0,0 +1,114 @@
+package com.leveluptasks.entity;
+
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "user")
+public class User {
+
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(unique = true, nullable = false)
+    private String firstname;
+    @Column(unique = true, nullable = false)
+    private String lastname;
+    @Column(unique = true, nullable = false)
+    private String email;
+    private String password;
+    private String profileImage;
+    private int level;
+    private int rewardPercentage;
+    @Enumerated(EnumType.STRING)
+    private Role role;
+
+
+    public enum Role {
```

Comment:
why not create a separate file for the enum ? so the class user only handle the user Entity

Reply:
for the Enum in Java I always use it in the class itself unlike Angular

Reply:
It is fine as long the enum is only used inside this class